### DISCUSSION TITLE
Strange buttons in native file-chooser

### DISCUSSION THREAD

**Discussion by abau** [fltk-rs/fltk-rs#1173] [Q&A] [answered]
When running the `editor` example in the `fltk-rs` repo, the native file-chooser has two strange buttons labeled `<--->` and `>---<` that are not present in the ` nativefilechooser-simple`  example in the `fltk` repository.

`fltk-rs`:

![rust](https://user-images.githubusercontent.com/4090497/161440539-1413cbe4-ecc6-40ed-80c8-1b6c85d8d84e.png)

`fltk`:

![cpp](https://user-images.githubusercontent.com/4090497/161440538-007deace-fb17-47f0-89cc-f9a59c2cfaf5.png)

How can I get rid of these two buttons?

My system:

- Operating System: Arch Linux
- KDE Plasma Version: 5.24.4
- KDE Frameworks Version: 5.92.0
- Qt Version: 5.15.3
- Kernel Version: 5.17.1-arch1-1 (64-bit)
- Graphics Platform: X11

**Comment by MoAlyousef** [comment, marked as answer]
fltk-rs (like recent FLTK) defaults to using kdialog on KDE desktops. FLTK offers a flag OPTION_USE_KDIALOG which can be turned off. Currently this flag isn’t exposed in fltk-rs, but I think I’ll expose it via a feature flag.

**Reply by abau** [reply]
I don't have `kdialog` installed. After installing it, the file-chooser in both sample applications look the same. So, it does not appear to have something to do with `kdialog`.

**Reply by MoAlyousef** [reply]
I’ll check tomorrow on a kubuntu system and get back to you.

**Reply by MoAlyousef** [reply]
This is from my kubuntu system:
![image](https://user-images.githubusercontent.com/37966791/161522197-98bec8ab-427a-4ab7-a531-0044ceda018b.png)

**Reply by MoAlyousef** [reply]
These buttons are from FLTK's Gtk native file chooser:
https://github.com/fltk/fltk/blob/2ee27e39db4e1e23cf9e917b671c7581be90b699/src/Fl_Native_File_Chooser_GTK.cxx#L756

If you're not using fltk-bundled, and you have kdialog installed, it should choose kdialog when you spawn a native file chooser.

**Reply by abau** [reply]
> This is from my kubuntu system:

I get the same when `kdialog` is installed.

> These buttons are from FLTK's Gtk native file chooser:

So, when I don't have `kdialog` installed, there is no way getting rid of these buttons, right?

**Reply by MoAlyousef** [reply]
Unfortunately that's right. You can request that they be removed from FLTK in the FLTK issue tracker or in the mailing list:
https://groups.google.com/g/fltkgeneral

**Reply by abau** [reply]
Still, what I don't understand is, why I do not get these buttons when compiling and running FLTK's own `nativefilechooser-simple` example. Maybe because of this [`Fl_Image::register_images_done` condition](https://github.com/fltk/fltk/blob/2ee27e39db4e1e23cf9e917b671c7581be90b699/src/Fl_Native_File_Chooser_GTK.cxx#L745).

**Reply by MoAlyousef** [reply]
What version of FLTK are you using, if not master, can you try building with FLTK master?

**Reply by abau** [reply]
`master` on commit `228d04d0e0d6e968c59bcab606e609eb1103f917`.

**Reply by MoAlyousef** [reply]
I've pushed a change when using the no-images features. Can you try:
```toml
[dependencies]
fltk = { git = "https://github.com/fltk-rs/fltk-rs", features = ["no-images"] }
```

Or if you're building after cloning this repo, try `git pull && cargo run --example editor --features=no-images`

**Reply by abau** [reply]
Using the `no-images` option hides these buttons. Very nice. Thank you!

I think, I will still ask on the FLTK mailing list if they could make these buttons configurable even when using images.

**Reply by abau** [reply]
Apparently, the GTK backend of the file chooser in FLTK now respects the `Preview` flag (cf. [here](https://groups.google.com/g/fltkgeneral/c/Z0l_IcVsoQk)).

**Reply by MoAlyousef** [reply]
Thanks for updating me. The next release should pull the fix from upstream.
